parser/transcript/src/main/java/de/danoeh/antennapod/parser/transcript/SrtTranscriptParser.java

```diff
@@ -0,0 +1,126 @@
+        long endTimecode = -1L;
+        long duration = 0L;
+
+        while (iter.hasNext()) {
+            line = iter.next();
+
+            if (line.isEmpty()) {
+                continue;
+            }
+
+            if (line.contains("-->")) {
+                String[] timecodes = line.split("-->");
+                if (timecodes.length < 2) {
+                    continue;
+                }
+                startTimecode = parseTimecode(timecodes[0].trim());
+                endTimecode = parseTimecode(timecodes[1].trim());
+                if (startTimecode == -1 || endTimecode == -1) {
+                    continue;
+                }
+
+                if (spanStartTimecode == -1) {
+                    spanStartTimecode = startTimecode;
+                }
+                duration += endTimecode - startTimecode;
+                do {
+                    line = iter.next();
+                    if (StringUtil.isBlank(line)) {
+                        break;
+                    }
+                    body.append(line.strip());
+                    body.append(" ");
+                } while (iter.hasNext());
+            }
+
+            if (body.indexOf(":") != -1) {
+                String [] parts = body.toString().trim().split(":");
+                if (parts.length < 2) {
+                    continue;
+                }
+                speaker = parts[0];
+                body = new StringBuffer(parts[1].strip());
+            }
+            if (!StringUtil.isBlank(body.toString())) {
+                segmentBody += " " + body;
+                segmentBody = StringUtils.trim(segmentBody);
+                if (duration >= minSpan && endTimecode > spanStartTimecode) {
+                    Log.d(TAG, "SRT : " + Long.toString(spanStartTimecode) + " " + segmentBody);
+                    transcript.addSegment(new TranscriptSegment(spanStartTimecode,
+                            endTimecode,
+                            segmentBody,
+                            speaker));
+                    duration = 0L;
+                    spanStartTimecode = -1L;
+                    segmentBody = "";
+                }
+                body = new StringBuffer("");
+            }
+        }
+
+        if (! StringUtil.isBlank(segmentBody) && endTimecode > spanStartTimecode) {
+            Log.d(TAG, "SRT : " + Long.toString(spanStartTimecode) + " " + segmentBody);
+
+            segmentBody = StringUtils.trim(segmentBody);
+            transcript.addSegment(new TranscriptSegment(spanStartTimecode,
+                    endTimecode,
+                    segmentBody,
+                    speaker));
+        }
+        if (transcript.getSegmentCount() > 0) {
+            return transcript;
+        } else {
+            return null;
+        }
+    }
+
+    // Time format 00:00:00,000
+    static long parseTimecode(String timecode) {
+        Pattern pattern = Pattern.compile("^([0-9]{2}):([0-9]{2}):([0-9]{2}),([0-9]{3})$");
```

Comment:
Could this be a member variable? Then the regex doesn't need to be parsed for every timecode. In theory, it could even be static, I guess, but maybe we don't want too many static initializations in many classes